src/pipeline/rackhd/docker/Publish.groovy

```diff
@@ -0,0 +1,59 @@
+package pipeline.rackhd.docker
+
+/*
+publish rackhd
+:library_dir: the directory of on-build-config
+:docker_tar_path: where the docker image tar lies
+:dockerhub_user: the rackhd user of dockerhub
+:dockerhub_pass: the rackhd user password of dockerhub
+:docker_record_path: where the docker build record file lies
+*/
+def publish(String library_dir, String docker_tar_path, String dockerhub_user, String dockerhub_pass, String docker_record_path) {
+	withCredentials([
+        usernamePassword(credentialsId: 'rackhd-ci-docker-hub',
+                         passwordVariable: 'DOCKERHUB_PASS',
+                         usernameVariable: 'DOCKERHUB_USER')]) {
```

Comment:
They are not used.
The user and password are passed by arguments

Reply:
got it. Thanks.